airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -129,10 +129,11 @@ def _group_logs_by_host(logs):
             key = getattr(log, 'host', 'default_host')
             grouped_logs[key].append(log)
 
-        # return items sorted by timestamp.
-        result = sorted(grouped_logs.items(), key=lambda kv: getattr(kv[1][0], 'message', '_'))
+        # return items sorted by asctime.
+        for host in grouped_logs:
+            grouped_logs[host].sort(key=lambda log: getattr(log, 'asctime', '_'))
```

Comment:
```suggestion
        def sorter(log):
            return getattr(log, 'asctime', '_')

        for logs in grouped_logs.values():
            logs.sort(key=sorter)
```

Repeated lambda definition inside a loop is slow.